### PR TITLE
Add the structures for the debugger

### PR DESCRIPTION
**Pull request by polaroi8d**
Add the structures what the debugger will be use. 

JerryScript-DCO-1.0-Signed-off-by: Levente Orban orbanl@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/debugger/debugger.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+{
+  package_type_t type;        /**< type of the package */
+  size_t msg_len;             /**< the lenght of the literal array */
+  char msg[128];              /**< the literal array */
+  uint32_t  line_offset;      /**< line offset */
+  uint32_t  line_index;       /**< line index */
+  uint16_t count;             /**< line count */
+} package;
+
+#endif /* JERRY_DEBUG_H */
```
Comment (zherczeg) [inline jerry-core/debugger/debugger.h]: Should be /\* JERRY_DEBUGGER_H */

File: jerry-core/debugger/debugger.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/* Copyright 2016 University of Szeged.
```
Comment (zherczeg) [inline jerry-core/debugger/debugger.h]: The file name should be jerry-debug.h

Comment (zherczeg) [reply jerry-core/debugger/debugger.h]: Or jerry-debugger.h

File: jerry-core/debugger/debugger.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRY_DEBUGGER_H
+#define JERRY_DEBUGGER_H
```
Comment (zherczeg) [inline jerry-core/debugger/debugger.h]: This must come from the file name.

File: jerry-core/debugger/debugger.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+} package_type_t;
+
+typedef struct 
+{
+  package_type_t type;        /**< type of the package */
+  size_t msg_len;             /**< the lenght of the literal array */
+  char msg[128];              /**< the literal array */
+  uint32_t  line_offset;      /**< line offset */
+  uint32_t  line_index;       /**< line index */
+  uint16_t count;             /**< line count */
```
Comment (zherczeg) [inline jerry-core/debugger/debugger.h]: There should not be packed everything into one structure. The layout should look like this (without comments):

```
typdef struct
{
  uint8_t type;
  uint8_t size;
} jerry_debug_message_header;
```

And each message should have its own type.

```
typdef struct
{
  jerry_debug_message_header header;
  char file_name[1];
} jerry_debug_message_source_name;
```

One structure for each message.

Perhaps SOURCE_FILE type is enough for this patch.

File: jerry-core/debugger/debugger.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+typedef enum 
+{
+  SOURCE_FILE,                /**< name of the source file */
+  FUNCTION_NAME,              /**< name of the parsed function */
+  BREAKPOINT_INFO,            /**< breakpoint */
+} package_type_t;
+
+typedef struct 
+{
+  package_type_t type;        /**< type of the package */
```
Comment (zherczeg) [inline jerry-core/debugger/debugger.h]: We never use types with unspecified length for network messages because the compiler might use types for them.

**Comment by polaroi8d** [comment]
Updated the PR.

File: jerry-core/debugger/jerry-debugger.h [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  uint8_t size;                     /**< size of the message */
+} jerry_debug_message_header_t;
+
+/**
+* Source file name
+*/
+typdef struct
+{
+  jerry_debug_message_header header;    /**< header of the source file name struct */
+  char file_name[128];                  /**< the message */
```
Comment (zherczeg) [inline jerry-core/debugger/jerry-debugger.h]: The file name should be 1 character long, and the real name should adapt the size

File: jerry-core/debugger/jerry-debugger.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  uint8_t size;                     /**< size of the message */
+} jerry_debug_message_header_t;
+
+/**
+* Source file name
+*/
+typdef struct
+{
+  jerry_debug_message_header header;    /**< header of the source file name struct */
+  char file_name[1];                    /**< the message */
```
Comment (stampho) [inline jerry-core/debugger/jerry-debugger.h:36]: This is definetely wrong. This means 1 character long string.

**Review by stampho — CHANGES_REQUESTED** [review]

File: jerry-core/debugger/jerry-debugger.h (hunk 7 above)
Comment (zherczeg) [reply jerry-core/debugger/jerry-debugger.h:36]: No, this is correct.

This struct is a mapping struct, mapped to a buffer, and the maximum buffer size limits the file_name.

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**